ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+    # the process, keep a count of the number of obs files that already exist
+    # on disk.
+    num_existing_files = 0
+    do_break = False
+    for fg in field_groups_in_obs:
+        for yyyymmddhh, fp_proc in zip(obs_retrieve_times_crnt_day, all_fp_proc_dict[fg]):
+            # Check whether the processed file already exists.
+            if os.path.isfile(fp_proc):
+                num_existing_files += 1
+                msg = dedent(f"""
+                    File already exists on disk:
+                        fp_proc = {fp_proc}
+                    """)
+                logging.info(msg)
+            else:
+                hod = yyyymmddhh.hour
+                arcv_hr_start = get_obs_arcv_hr(obtype, arcv_intvl_hrs, hod)
+                msg = dedent(f"""
+                    File does not exist on disk:
+                        fp_proc = {fp_proc}
+                    Setting the hour (since hour 0 of the current task day) of the first
+                    archive to retrieve to:
+                        arcv_hr_start = {arcv_hr_start}
+                    """)
+                logging.info(msg)
+                do_break = True
+                break
+        if do_break: break
+
+    # If the number of obs files that already exist on disk is equal to the
+    # number of obs files needed, then there is no need to retrieve any files.
+    num_obs_retrieve_times_crnt_day = len(obs_retrieve_times_crnt_day)
+    num_files_needed = num_obs_retrieve_times_crnt_day*num_field_groups
+    if num_existing_files == num_files_needed:
+
+        msg = dedent(f"""
+            All obs files needed for the current day (yyyymmdd_task) already exist
+            on disk:
+                yyyymmdd_task = {yyyymmdd_task}
+            Thus, there is no need to retrieve any files.
+            """)
+        logging.info(msg)
+        return True
+
+    # If the number of obs files that already exist on disk is not equal to
+    # the number of obs files needed, then we will need to retrieve files.
+    # In this case, set the sequence of hours corresponding to the archives
+    # from which files will be retrieved.
+    else:
+
+        arcv_hrs = [hr for hr in range(arcv_hr_start, arcv_hr_end+arcv_intvl_hrs, arcv_intvl_hrs)]
+        msg = dedent(f"""
+            At least some obs files needed needed for the current day (yyyymmdd_task)
+            do not exist on disk:
+                yyyymmdd_task = {yyyymmdd_task}
+            The number of obs files needed for the current day is:
+                num_files_needed = {num_files_needed}
+            The number of obs files that already exist on disk is:
+                num_existing_files = {num_existing_files}
+            Will retrieve remaining files by looping over archives corresponding to
+            the following hours (since hour 0 of the current day):
+                arcv_hrs = {arcv_hrs}
+            """)
+        logging.info(msg)
```

Comment:
Since the `if` condition ends in a `return` statement, this `else` is unnecessary
```suggestion
    arcv_hrs = [hr for hr in range(arcv_hr_start, arcv_hr_end+arcv_intvl_hrs, arcv_intvl_hrs)]
    msg = dedent(f"""
        At least some obs files needed needed for the current day (yyyymmdd_task)
        do not exist on disk:
            yyyymmdd_task = {yyyymmdd_task}
        The number of obs files needed for the current day is:
            num_files_needed = {num_files_needed}
        The number of obs files that already exist on disk is:
            num_existing_files = {num_existing_files}
        Will retrieve remaining files by looping over archives corresponding to
        the following hours (since hour 0 of the current day):
            arcv_hrs = {arcv_hrs}
        """)
    logging.info(msg)
```

Reply:
Done.